### DISCUSSION TITLE
Unban and add to buddy list

### DISCUSSION THREAD

**Discussion by vkinsella72** [nicotine-plus/nicotine-plus#2426] [Q&A] [answered]
I have an autoban plugin. But there's a problem. If a user is on my buddy list; how do I prevent them from being banned, even if the number of files they share is below the required amount?

**Comment by slook** [comment, marked as answer]
Use this condition to bail out somewhere before the code that does the ban:

https://github.com/nicotine-plus/nicotine-plus/blob/30d4f29a0fdc2795e69cd3c716f138bb429a5b4d/pynicotine/plugins/leech_detector/__init__.py#L110-L114

**Reply by vkinsella72** [reply]
Sweet ❤️ I can see how that would resolve it, thank you x

**Reply by vkinsella72** [reply]
As a side note, how would I make that conditional; with a switch in the plugin interface? I know I may be asking a lot; however I haven't done much programming in the last twenty years; so my skills are a tad rusty 🙈 x

**Reply by slook** [reply]
The settings are defined in `__init__()`...

Add a setting to `self..settings` with a default value such as:
`            "allow_buddies": True,`

Add a corresponding item into `self.metasettings` how you want to appear in the settings dialog:
```
            "allow_buddies": {
                "description": "Allow buddies to leech:",
                "type": "bool"
            },
```

Then change the condition to:

`    if self.settings["allow_buddies"] and user in self.core.userlist.buddies: `

To reload the plugin after making your changes, just disable it and enable it again, it is not needed to restart Nicotine+.

To view any errors thrown by the Python interpreter, enable Log View Pane > right click > Log Categories > `[Debug] Miscellaneous`. If there are any errors in your code, then the plugin won't load and you will see which line is the problem in the debug log.

**Reply by vkinsella72** [reply]
Absolutely amazing! And it worked 😊 Thank you very, very much ❤️❤️ x
